sw/device/silicon_creator/lib/BUILD

```diff
@@ -0,0 +1,431 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan.bzl", "OPENTITAN_CPU", "opentitan_binary", "opentitan_functest", "verilator_params")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "boot_data_intf",
+    hdrs = ["boot_data.h"],
+    deps = [
+        ":error",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/drivers:hmac",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+    ],
+)
+
+cc_library(
+    name = "boot_data",
+    srcs = ["boot_data.c"],
+    deps = [
+        ":boot_data_intf",
+        ":error",
+        "//hw/ip/flash_ctrl/data:flash_ctrl_regs",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/base",
+        "//sw/device/silicon_creator/lib/drivers:flash_ctrl",
+        "//sw/device/silicon_creator/lib/drivers:hmac",
+    ],
+)
+
+cc_library(
+    name = "mock_boot_data",
+    testonly = True,
+    hdrs = ["mock_boot_data.h"],
+    deps = [
+        ":boot_data_intf",
+        "//sw/device/silicon_creator/testing:mask_rom_test",
+        "@googletest//:gtest",
+    ],
+)
+
+opentitan_functest(
+    name = "boot_data_functest",
+    srcs = ["boot_data_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+    ),
+    deps = [
+        ":boot_data",
+        ":test_main",
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/silicon_creator/lib/drivers:flash_ctrl",
+    ],
+)
+
+cc_library(
+    name = "epmp_intf",
+    hdrs = [
+        "epmp.h",
+        "epmp_defs.h",
+    ],
+    deps = [
+        ":error",
+    ],
+)
+
+cc_library(
+    name = "epmp",
+    srcs = ["epmp.c"],
+    target_compatible_with = [OPENTITAN_CPU],
+    deps = [
+        ":epmp_intf",
+        ":error",
+        "//sw/device/lib/base",
+    ],
+)
+
+cc_test(
+    name = "epmp_unittest",
+    srcs = [
+        "epmp_unittest.cc",
+        # sources included here because we need MOCK_CSR.
```

Comment:
This seems like a bug. when building `:epmp` for the host it should automatically turn this flag on.

Reply:
Similar to [this comment](https://github.com/lowRISC/opentitan/pull/9412/files/8e1d1090a377bcd85949d7ba099a28d28517cf4a#r763096560), this is consistent with the meson rules.

Personally, I see this as a post-bazel-conversion refactoring.

Reply:
I mean, this is your refactor, but I think that's actually a bug in the meson rules, for the simple reason that meson rules cannot be polymorphic on target; because we were lazy (and fighting meson is a poor use of engineering time), we didn't bother defining on-host, test versions of things.

I would be ok letting this go as long as we file an issue along the lines of "we should have a single IS_HOST_TEST macro that bazel sets", which I think can coexist with meosn if we do something like
```c
#ifdef MOCK_CSR
#define OT_IS_HOST_TEST
#endif
```
which we then plan to delete later.

I mostly say this because I'm worried about this getting forgotten about.

Reply:
Added items to #7972.